upb/decode_fast.c

```diff
@@ -396,61 +350,63 @@ static const char *fastdecode_submsg(UPB_PARSE_PARAMS, int tagbytes,
         end = (void*)(d->arena_ptr + (new_size * sizeof(upb_msg*)));
         d->arena_ptr += new_bytes;
       } else {
-        d->limit = saved_limit;
-        d->fastlimit = saved_fastlimit;
-        d->depth++;
-        RETURN_GENERIC("need array realloc\n");
+        goto repeated_generic;
```

Comment:
`arr` is NULL in this branch. This is jumping to code that writes to `arr->len`. Segv?

Reply:
oh right. Okay I hoisted this out of the loop. I think that is less confusing, now it looks like it can happen more